src/mpi/init/initthread.c

```diff
@@ -140,6 +140,16 @@ int MPIR_Init_thread(int *argc, char ***argv, int required, int *provided)
     mpi_errno = MPII_init_async(thread_provided);
     MPIR_ERR_CHECK(mpi_errno);
 
+    /* connect to remote processes is has parent */
+    if (MPIR_Process.has_parent) {
+        char parent_port[MPI_MAX_PORT_NAME];
+        mpi_errno = MPIR_pmi_get_parent_port(parent_port);
+        MPIR_ERR_CHECK(mpi_errno);
+        MPID_Comm_connect(parent_port, NULL, 0, MPIR_Process.comm_world, &MPIR_Process.comm_parent);
+        MPIR_Assert(MPIR_Process.comm_parent != NULL);
+        MPL_strncpy(MPIR_Process.comm_parent->name, "MPI_COMM_PARENT", MPI_MAX_OBJECT_NAME);
+    }
```

Comment:
@yfguo How about wrap this block of code into `MPID_spawn_init()`, then I wonder if we could also wrap the relevant code in ch3 into `MPID_spawn_init()`? If yes, the problem will be solved, right?

Reply:
That would certainly solve the problem. But I don't like the idea of doing that. Let me see how can I fix CH3 code to use `MPIR_pmi_`.

Reply:
Hmm, on second thought, that is not so bad comparing to the current version which makes `MPIR_Init_thread` explicity depends on `MPIR_pmi_`.

Reply:
Currently seems all spawn related code are in MPID layer, so might as well keep them inside MPID layer.